### PR TITLE
Add prometheus-to-sd to scrape metrics

### LINKED ISSUE #976: Visualise prometheus metrics on GCP
Subtask of #736:

Cloud Monitoring recommends using [OpenCensus for collecting custom metrics](https://cloud.google.com/monitoring/custom-metrics/open-census). However, there don't seem to be a client library for Rust. Furthermore, since we already use prometheus to collect metrics, using a client library for OpenCensus sounds redundant. 

Alternatively, we could deploy an OpenCensus server (or configure it accordingly if one is by default available for Kubernetes). OpenCensus can be configured to [receive prometheus metrics](https://opencensus.io/service/receivers/prometheus/) (also see [full config](https://github.com/prometheus/prometheus/blob/master/documentation/examples/prometheus.yml) for a prometheus server), and to [export them to stackdriver](https://opencensus.io/service/exporters/stackdriver/metrics/) (which should allow viewing the metrics in GCP).

The other alternative, which might be the simplest way to export prometheus metrics to GCP, is to use the [Stackdriver Prometheus sidecar](https://github.com/Stackdriver/stackdriver-prometheus-sidecar). 

Another alternative: [prometheus-to-sd](https://github.com/GoogleCloudPlatform/k8s-stackdriver/tree/master/prometheus-to-sd) is a simple component that can scrape metrics stored in prometheus text format.

### PR DESCRIPTION
**Pull request by rbehjati**
Added a new container from [prometheus-to-sd](https://github.com/GoogleCloudPlatform/k8s-stackdriver/tree/master/prometheus-to-sd)  to the aggregator pod. This container scrapes metrics from the aggregator oak-application.

Fixes #976 

# Checklist

- [ ] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by [Cloudbuild](cloudbuild.yaml)
  - [ ] I have updated [documentation](docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

File: examples/aggregator/gcp/pod.yaml [hunk 1]
```diff
@@ -24,6 +27,26 @@ spec:
       env:
         - name: RUST_LOG
           value: info
+    - name: prometheus-to-sd
+      image: gcr.io/google-containers/prometheus-to-sd:v0.9.0
+      ports:
+        - name: profiler
```
Comment (tiziano88) [inline examples/aggregator/gcp/pod.yaml:33]: What does `profiler` refer to?

File: examples/aggregator/gcp/pod.yaml [hunk 2]
```diff
@@ -24,6 +27,26 @@ spec:
...
+          containerPort: 6060
+      command:
+        - /monitor
+        - --stackdriver-prefix=custom.googleapis.com
```
Comment (tiziano88) [inline examples/aggregator/gcp/pod.yaml:37]: Does this need to be this exact value? If so, could you add a link to some source?

File: examples/aggregator/gcp/pod.yaml [hunk 3]
```diff
@@ -24,6 +27,26 @@ spec:
       env:
         - name: RUST_LOG
           value: info
+    - name: prometheus-to-sd
```
Comment (tiziano88) [inline examples/aggregator/gcp/pod.yaml:30]: Could you add a comment here? in particular, the `sd` part of the name is quite obscure. The link you have in the commit message should probably be here instead.

File: examples/aggregator/gcp/pod.yaml [hunk 4]
```diff
@@ -24,6 +27,26 @@ spec:
...
+      image: gcr.io/google-containers/prometheus-to-sd:v0.9.0
+      ports:
+        - name: profiler
+          containerPort: 6060
+      command:
+        - /monitor
+        - --stackdriver-prefix=custom.googleapis.com
+        - --source=oak-application:http://localhost:3030/metrics
```
Comment (tiziano88) [inline examples/aggregator/gcp/pod.yaml:38]: Does this support a single replica of a single source? How will it work with a more realistic deployment?

File: examples/aggregator/server/BUILD [hunk 5]
```diff
@@ -26,6 +26,7 @@ oak_docker(
     application = "//examples/aggregator/config:config",
     ports = [
         "8080/tcp",  # Oak Application gRPC port.
+        "3030/tcp",  # Metrics port.
```
Comment (ipetr0v) [inline examples/aggregator/server/BUILD:29]: Does it also need the `6060` port?

File: examples/aggregator/gcp/pod.yaml [hunk 6]
```diff
@@ -24,6 +27,30 @@ spec:
...
+    # https://github.com/GoogleCloudPlatform/k8s-stackdriver/tree/master/prometheus-to-sd
+    - name: prometheus-to-stackdriver
+      image: gcr.io/google-containers/prometheus-to-sd:v0.9.0
+      ports:
+        - containerPort: 6060
+      command:
+        - /monitor
+        # To be able to push custom metrics to the Stackdriver, the `stackdriver-prefix` 
+        # flag must be set to `custom.googleapis.com`.
+        # Ref: https://github.com/GoogleCloudPlatform/k8s-stackdriver/tree/master/prometheus-to-sd#custom-metrics
```
Comment (tiziano88) [inline examples/aggregator/gcp/pod.yaml:40]: Perhaps also add this https://cloud.google.com/monitoring/custom-metrics#about_custom_metrics which is the actual GCP reference.

File: examples/aggregator/gcp/pod.yaml [hunk 7]
```diff
@@ -24,6 +27,30 @@ spec:
       env:
         - name: RUST_LOG
           value: info
+    # Additional container to scrape prometheus metrics:
+    # https://github.com/GoogleCloudPlatform/k8s-stackdriver/tree/master/prometheus-to-sd
+    - name: prometheus-to-stackdriver
+      image: gcr.io/google-containers/prometheus-to-sd:v0.9.0
```
Comment (tiziano88) [inline examples/aggregator/gcp/pod.yaml:33]: FYI it looks like the latest version is 0.9.2, is there any reason why it was not chosen?

See https://pantheon.corp.google.com/gcr/images/google-containers/GLOBAL/prometheus-to-sd?gcrImageListsize=30

**Comment by tiziano88** [comment]
Note from https://github.com/GoogleCloudPlatform/k8s-stackdriver/tree/master/prometheus-to-sd#alternatives :

> Google develops prometheus-to-sd primarily for Google Kubernetes Engine to collect metrics from system services in order to support Kubernetes users. We designed the tool to be lean when deployed as a sidecar in your pod. It's intended to support only the metrics the Kubernetes team at Google needs; you can use it to monitor your applications, but the feature set is limited.

@rbehjati it seems we should be using https://github.com/Stackdriver/stackdriver-prometheus-sidecar instead?

**Comment by rbehjati** [comment]
> Note from https://github.com/GoogleCloudPlatform/k8s-stackdriver/tree/master/prometheus-to-sd#alternatives :
> 
> > Google develops prometheus-to-sd primarily for Google Kubernetes Engine to collect metrics from system services in order to support Kubernetes users. We designed the tool to be lean when deployed as a sidecar in your pod. It's intended to support only the metrics the Kubernetes team at Google needs; you can use it to monitor your applications, but the feature set is limited.
> 
> @rbehjati it seems we should be using https://github.com/Stackdriver/stackdriver-prometheus-sidecar instead?
> 
> also see https://cloud.google.com/monitoring/kubernetes-engine/prometheus

Yes. I can try `stackdriver-prometheus-sidecar` next. For that I think we have to have a deployment (instead of just a pod). I can fix that.

**Pull request merged**